keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py

```diff
@@ -0,0 +1,633 @@
+    yolo_v8_detector_presets,
+)
+from keras_cv.models.object_detection.yolo_v8.yolo_v8_iou_loss import (
+    YOLOV8IoULoss,
+)
+from keras_cv.models.object_detection.yolo_v8.yolo_v8_label_encoder import (
+    YOLOV8LabelEncoder,
+)
+from keras_cv.models.object_detection.yolo_v8.yolo_v8_layers import (
+    apply_conv_bn,
+)
+from keras_cv.models.object_detection.yolo_v8.yolo_v8_layers import (
+    apply_csp_block,
+)
+from keras_cv.models.task import Task
+from keras_cv.utils.python_utils import classproperty
+from keras_cv.utils.train import get_feature_extractor
+
+BOX_REGRESSION_CHANNELS = 64
+
+
+def get_anchors(
+    image_shape,
+    strides=[8, 16, 32],
+    base_anchors=[0.5, 0.5],
+):
+    """Gets anchor points for YOLOV8.
+
+    YOLOV8 uses anchor points representing the center of proposed boxes, and
+    matches ground truth boxes to anchors based on center points.
+
+    Args:
+        image_shape: tuple or list of two integers representing the heigh and
+            width of input images, respectively.
+        strides: tuple of list of integers, the size of the strides across the
+            image size that should be used to create anchors.
+        base_anchors: tuple or list of two integers representing the offset from
+            (0,0) to start creating the center of anchor boxes, releative to the
+            stride. For example, using the default (0.5, 0.5) creates the first
+            anchor box for each stride such that its center is half of a stride
+            from the edge of the image.
+
+    Returns:
+        A tuple of anchor centerpoints and anchor strides. Multiplying the
+        two together will yield the centerpoints in absolute x,y format.
+
+    """
+    base_anchors = tf.constant(base_anchors, dtype=tf.float32)
+
+    all_anchors = []
+    all_strides = []
+    for stride in strides:
+        hh_centers = tf.range(start=0, limit=image_shape[0], delta=stride)
+        ww_centers = tf.range(start=0, limit=image_shape[1], delta=stride)
+        ww_grid, hh_grid = tf.meshgrid(ww_centers, hh_centers)
+        grid = tf.cast(
+            tf.reshape(tf.stack([hh_grid, ww_grid], 2), [-1, 1, 2]),
+            tf.float32,
+        )
+        anchors = tf.expand_dims(base_anchors * [stride, stride], 0) + grid
+        anchors = tf.reshape(anchors, [-1, 2])
+        all_anchors.append(anchors)
+        all_strides.append(tf.repeat(stride, anchors.shape[0]))
+
+    all_anchors = tf.cast(tf.concat(all_anchors, axis=0), tf.float32)
+    all_strides = tf.cast(tf.concat(all_strides, axis=0), tf.float32)
+
+    all_anchors = all_anchors / all_strides[:, None]
+
+    # Swap the x and y coordinates of the anchors.
+    all_anchors = tf.concat(
+        [all_anchors[:, 1, tf.newaxis], all_anchors[:, 0, tf.newaxis]], axis=-1
+    )
+    return all_anchors, all_strides
+
+
+def apply_path_aggregation_fpn(features, depth=3, name="fpn"):
+    """
```

Comment:
Nit: docstrings should start with a one line description on the same line as the opening triple quotes.

Reply:
Done